strict_encoding/src/monero.rs

```diff
@@ -0,0 +1,258 @@
+}
+
+impl Strategy for RangeSig {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for RctSigBase {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Signature {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for VarInt {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl StrictEncode for Amount {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.as_pico().strict_encode(&mut e)
+    }
+}
+
+impl StrictDecode for Amount {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        Ok(Self::from_pico(u64::strict_decode(&mut d)?))
+    }
+}
+
+impl StrictEncode for SignedAmount {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.as_pico().strict_encode(&mut e)
+    }
+}
+
+impl StrictDecode for SignedAmount {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        Ok(Self::from_pico(i64::strict_decode(&mut d)?))
+    }
+}
+
+impl StrictEncode for PaymentId {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        Ok(e.write(&self.to_fixed_bytes())?)
+    }
+}
+
+impl StrictDecode for PaymentId {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        let mut bytes = [0u8; 8];
+        d.read_exact(&mut bytes)?;
+        Ok(Self::from(bytes))
+    }
+}
+
+impl StrictEncode for Index {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.major.strict_encode(&mut e)?;
+        self.minor.strict_encode(e)
+    }
+}
+
+impl StrictDecode for Index {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        let major = u32::strict_decode(&mut d)?;
+        let minor = u32::strict_decode(d)?;
+        Ok(Self { major, minor })
+    }
+}
+
+impl StrictEncode for KeyPair {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.view.strict_encode(&mut e)?;
+        self.spend.strict_encode(e)
+    }
+}
+
+impl StrictDecode for KeyPair {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        let view = PrivateKey::strict_decode(&mut d)?;
+        let spend = PrivateKey::strict_decode(d)?;
+        Ok(Self { view, spend })
+    }
+}
+
+impl StrictEncode for ViewPair {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.view.strict_encode(&mut e)?;
+        self.spend.strict_encode(e)
+    }
+}
+
+impl StrictDecode for ViewPair {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        let view = PrivateKey::strict_decode(&mut d)?;
+        let spend = PublicKey::strict_decode(d)?;
+        Ok(Self { view, spend })
+    }
+}
```

Comment:
Here you can also use macros

Reply:
Done in f7d17021bd44ec9630adbca916bf341865866079